### PR TITLE
DR-2540 - Utilize GCSConfiguration rather than constant for timeout values

### PR DESCRIPTION
**Pull request by snf2ye**

### CODE REVIEW

File: src/main/java/bio/terra/service/tabulardata/google/BigQueryProject.java [hunk 1]
```diff
@@ -33,23 +34,32 @@
...
     logger.info("Retrieving Bigquery project for project id: {}", projectId);
     this.projectId = projectId;
     HttpTransportOptions transportOptions = StorageOptions.getDefaultHttpTransportOptions();
+    GcsConfiguration gcsConfiguration = appContext.getBean(GcsConfiguration.class);
     transportOptions =
         transportOptions.toBuilder()
-            .setConnectTimeout(TIMEOUT_SECONDS * 1000)
-            .setReadTimeout(TIMEOUT_SECONDS * 1000)
+            .setConnectTimeout(gcsConfiguration.getConnectTimeoutSeconds() * 1000)
+            .setReadTimeout(gcsConfiguration.getReadTimeoutSeconds() * 1000)
```
Comment (tlangs) [inline src/main/java/bio/terra/service/tabulardata/google/BigQueryProject.java]: Could condense this down into `int timeoutMillis =  gcsConfiguration.getConnectTimeoutSeconds() * 1000` and then just use that in place of the two calls to `gcsConfiguration`.

**Comment by tlangs** [comment]
I'm glad this was easier than we initially thought!

File: src/main/java/bio/terra/service/tabulardata/google/BigQueryProject.java (hunk 1 above)
Comment (snf2ye) [reply src/main/java/bio/terra/service/tabulardata/google/BigQueryProject.java]: They look very similar and they actually have the same value in app config, but these are two settings (connect & read timeouts).

File: src/main/java/bio/terra/service/tabulardata/google/BigQueryProject.java [hunk 2]
```diff
@@ -33,23 +35,25 @@
...
     logger.info("Retrieving Bigquery project for project id: {}", projectId);
     this.projectId = projectId;
     HttpTransportOptions transportOptions = StorageOptions.getDefaultHttpTransportOptions();
+    ApplicationContext appCtx = ApplicationContextUtils.getApplicationContext();
+    GcsConfiguration gcsConfiguration = appCtx.getBean(GcsConfiguration.class);
     transportOptions =
         transportOptions.toBuilder()
-            .setConnectTimeout(TIMEOUT_SECONDS * 1000)
-            .setReadTimeout(TIMEOUT_SECONDS * 1000)
+            .setConnectTimeout(gcsConfiguration.getConnectTimeoutSeconds() * 1000)
```
Comment (nmalfroy) [inline src/main/java/bio/terra/service/tabulardata/google/BigQueryProject.java]: I think for this kind of thing we typically would write it as:
```suggestion
            .setConnectTimeout(TimeUnit.SECONDS.toMillis(gcsConfiguration.getConnectTimeoutSeconds()))
```

File: src/main/java/bio/terra/app/configuration/ApplicationContextUtils.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  @SuppressFBWarnings(
+      value = "ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD",
+      justification =
+          "We need setAppContext to override method in ApplicationContextAware, but need rest of class to be static. Reason for spotbugs: 'This instance method writes to a static field. This is tricky to get correct if multiple instances are being manipulated, and generally bad practice.' Since we are using this specifically to pull a config value that should stay constant, this shouldn't be a concern.")
+  public void setApplicationContext(ApplicationContext appContext) {
+    ctx = appContext;
+  }
+
+  public static ApplicationContext getApplicationContext() {
+    return ctx;
```
Comment (nmalfroy) [inline src/main/java/bio/terra/app/configuration/ApplicationContextUtils.java:29]: Does it make sense to have a check here that sees if it's null and throws something like "the application context isn't initialized yet"?

**Review by nmalfroy — CHANGES_REQUESTED** [review]
Looks good!  Just wondering about putting in an extra check and to use the TimeUnit method for dealing with moving between time units

File: src/main/java/bio/terra/service/tabulardata/google/BigQueryProject.java (hunk 2 above)
Comment (snf2ye) [reply src/main/java/bio/terra/service/tabulardata/google/BigQueryProject.java]: Oh nice, that is better.

File: src/main/java/bio/terra/app/configuration/ApplicationContextUtils.java (hunk 3 above)
Comment (snf2ye) [reply src/main/java/bio/terra/app/configuration/ApplicationContextUtils.java:29]: Sounds good!

**Review by tlangs — APPROVED** [review]
Weird Spring magic!

**Review by nmalfroy — APPROVED** [review]
thanks for making the changes!

**Pull request merged**